pkg/kubernetes/persistentvolumeclaim/v1alpha1/build.go

```diff
@@ -0,0 +1,115 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// Builder is the builder object for PVC
+type Builder struct {
+	pvc  *PVC
+	errs []error
+}
+
+// NewBuilder returns new instance of Builder
+func NewBuilder() *Builder {
+	return &Builder{pvc: &PVC{object: &corev1.PersistentVolumeClaim{}}}
+}
+
+// WithName sets the Name field of PVC with provided value.
+func (b *Builder) WithName(name string) *Builder {
+	if len(name) == 0 {
+		b.errs = append(b.errs, errors.New("failed to build PVC object: missing PVC name"))
+		return b
+	}
+	b.pvc.object.Name = name
+	return b
+}
+
+// WithNamespace sets the Namespace field of PVC provided arguments
+func (b *Builder) WithNamespace(namespace string) *Builder {
+	if len(namespace) == 0 {
+		namespace = "default"
+	}
+	b.pvc.object.Namespace = namespace
+	return b
+}
+
+// WithAnnotations sets the Annotations field of PVC with provided arguments
+func (b *Builder) WithAnnotations(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errs = append(b.errs, errors.New("failed to build PVC object: missing annotations"))
+		return b
+	}
+	b.pvc.object.Annotations = annotations
+	return b
+}
+
+// WithLabels sets the Labels field of PVC with provided arguments
+func (b *Builder) WithLabels(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(b.errs, errors.New("failed to build PVC object: missing labels"))
+		return b
+	}
+	b.pvc.object.Labels = labels
+	return b
+}
+
+// WithStorageClass sets the StorageClass field of PVC with provided arguments
+func (b *Builder) WithStorageClass(scName string) *Builder {
+	if len(scName) == 0 {
+		b.errs = append(b.errs, errors.New("failed to build PVC object: missing storageclass name"))
+		return b
+	}
+	b.pvc.object.Spec.StorageClassName = &scName
+	return b
+}
+
+// WithAccessModes sets the AccessMode field in PVC with provided arguments
+func (b *Builder) WithAccessModes(accessMode []corev1.PersistentVolumeAccessMode) *Builder {
+	if len(accessMode) == 0 {
+		b.errs = append(b.errs, errors.New("failed to build PVC object: missing accessmodes"))
+		return b
+	}
+	b.pvc.object.Spec.AccessModes = accessMode
+	return b
+}
+
+// WithCapacity sets the Capacity field in PVC with provided arguments
+func (b *Builder) WithCapacity(capacity string) *Builder {
+	resCapacity, err := resource.ParseQuantity(capacity)
+	if err != nil {
+		b.errs = append(b.errs, errors.Errorf("failed to build PVC object: failed to parse capacity {%s}", capacity))
```

Comment:
We are missing the actual error.
errors.Wrapf() should be used.

Reply:
Updated with errors.Wrapf()